mongo-migration/cmd/migrate-postgres-to-mongo/lambda/main.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	postgresgateway "github.com/Pocket/global-services/mongo-migration/postgres_client"
+	"github.com/Pocket/global-services/shared/apigateway"
+	"github.com/Pocket/global-services/shared/database"
+	"github.com/Pocket/global-services/shared/gateway/models"
+	logger "github.com/Pocket/global-services/shared/logger"
+	"github.com/Pocket/global-services/shared/utils"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/pokt-foundation/portal-api-go/repository"
+	"github.com/pokt-foundation/utils-go/environment"
+)
+
+var (
+	postgresClientUrl     = environment.MustGetString("POSTGRES_CLIENT_URL")
+	authenticationToken   = environment.MustGetString("AUTHENTICATION_TOKEN")
+	mongoConnectionString = environment.MustGetString("MONGO_CONNECTION_STRING")
+	mongoDatabase         = environment.GetString("MONGO_DATABASE", "gateway")
+)
+
+// LambdaHandler manages the process of getting the postgres apps migrated to mongo
+func LambdaHandler(ctx context.Context) (events.APIGatewayProxyResponse, error) {
+	appsCount, lbsCount, err := migrateToMongo(ctx)
+
+	if err != nil {
+		return *apigateway.NewJSONResponse(http.StatusOK, map[string]any{
+			"err": err.Error(),
+		}), nil
+	}
+	return *apigateway.NewJSONResponse(http.StatusOK, map[string]any{
+		"newApps": appsCount,
+		"newLbs":  lbsCount,
+	}), nil
+}
+
+func main() {
+	lambda.Start(LambdaHandler)
+}
+
+func migrateToMongo(ctx context.Context) (int, int, error) {
+	mongo, err := database.ClientFromURI(ctx, mongoConnectionString, mongoDatabase)
+	if err != nil {
+		return 0, 0, errors.New("error connecting to mongo: " + err.Error())
+	}
+
+	postgres := postgresgateway.NewPostgresClient(postgresClientUrl, authenticationToken)
+
+	mongoApps, postgresApps, err := getApplications(ctx, mongo, postgres)
+	if err != nil {
+		return 0, 0, err
+	}
+	mongoLBs, postgresLBs, err := getLoadBalancers(ctx, mongo, postgres)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	appsToWrite := convertRepositoryToMongo(getItemsNotInMongo(postgresApps, mongoApps), models.RepositoryToModelApp)
```

Comment:
I think we need to compare the two lists to only add entries that are needed to Mongo (unless the intention is to consistently UPDATE and keep in exact sync with Postgres DB?

Reply:
yeah this only adds new entries once, that results an empty slice in case there is none